File: Simpleinherit.java
import java.util.Scanner;

class Parallelogram {
    public float diagonal1, diagonal2, angle, result;

    public float areaOfParallelogram(float d1, float d2, float angle) {
        float angle1 = (float) (angle * Math.PI / 180);
        result = 0.5f * (d1 * d2) * (float) Math.sin(angle1);
        return result;
    }
}

class Rectangle extends Parallelogram {
    public float length, breadth, result;

    public float areaOfRectangle(float length, float breadth) {
        result = length * breadth;
        return result;
    }
}

public class Simpleinherit {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Rectangle obj1 = new Rectangle();

        System.out.println("To find the area of Parallelogram");
        System.out.println("Enter lengths of two diagonals and angle between them: ");
        obj1.diagonal1 = scanner.nextFloat();
        obj1.diagonal2 = scanner.nextFloat();
        obj1.angle = scanner.nextFloat();
        obj1.result = obj1.areaOfParallelogram(obj1.diagonal1, obj1.diagonal2, obj1.angle);
        System.out.println("Area of parallelogram is " + obj1.result);

        System.out.println("To find the area of rectangle");
        System.out.println("Enter lengths of two length and breadth: ");
        obj1.length = scanner.nextFloat();
        obj1.breadth = scanner.nextFloat();
        obj1.result = obj1.areaOfRectangle(obj1.length, obj1.breadth);
        System.out.println("Area of rectangle is " + obj1.result);
        scanner.close();
    }
}

File: Simpleinherit.cpp
#include<iostream>
#include<cmath>
using namespace std;
class Parlleogram{
    public:
        float diagonal1,diagonal2,angle,result;
        float AreaofParlleogram(float d1,float d2,float angle){
            float angle1 = angle*3.14/180;
            result = 0.5*(d1*d2)*sin(angle1);
            return result;
        }
};

class Rectangle : public Parlleogram{
    public:
        float length,breadth,result;
        float AreaofRectangle(float length,float breadth){
            result = length*breadth;
            return result;
        }
};

int main(){
    Rectangle obj1;
    cout<<"To find area of Parlleogram"<<endl;
    cout<<"enter lengths of two diagonals and angle between them: "<<endl;
    cin>>obj1.diagonal1>>obj1.diagonal2>>obj1.angle;
    obj1.result=obj1.AreaofParlleogram(obj1.diagonal1,obj1.diagonal2,obj1.angle);
    cout<<"Area of parlleogram is "<<obj1.result<<endl;

    cout<<"To find area of rectangle"<<endl;
    cout<<"enter lengths of two length and breadth: "<<endl;
    cin>>obj1.length>>obj1.breadth;
    obj1.result=obj1.AreaofRectangle(obj1.length,obj1.breadth);
    cout<<"Area of rectangle is "<<obj1.result<<endl;

}
